controllers/storagesystem_controller.go

```diff
@@ -47,9 +51,11 @@ const (
 // StorageSystemReconciler reconciles a StorageSystem object
 type StorageSystemReconciler struct {
 	client.Client
-	Log      logr.Logger
-	Scheme   *runtime.Scheme
-	Recorder *EventReporter
+	Log               logr.Logger
+	Scheme            *runtime.Scheme
+	Recorder          *EventReporter
+	OperatorNamespace string
+	ctx               context.Context
```

Comment:
`ctx` is a pair request variable and should not be stored on the reconciler object which is shared between event. 

Bugs will only be visible when we will set up the controller concurrency to more then 1, but even if we never will, this is still a bug. As a matter of fact the person which will raise the concurrency will not even know that they are introducing a bug, as reproduction of the but will be depended on exact scheduling conditions. 

For data the is request scoped, please pass it through the function invocation as parameters (as separate values or as a compound struct), or alternatively follow the approach taken here: https://github.com/ceph/ceph-csi-operator/blob/main/internal/controller/driver_controller.go#L87-L101

Reply:
This controller anyway will be removed.

Reply:
anyway, i removed it.